debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresSignalBasedIncrementalSnapshotChangeEventSource.java

```diff
@@ -57,4 +68,155 @@ protected Table refreshTableSchema(Table table) throws SQLException {
         schema.refresh(jdbcConnection, table.id(), true);
         return schema.tableFor(table.id());
     }
+
+    @Override
+    protected String buildChunkQuery(Table table, int limit, Optional<String> additionalCondition) {
+        // Select query builder based on the configuration only on PostgreSQL Connector (incremental.snapshot.query.builder.type)
+        if (postgresConnectorConfig.getIncrementalSnapshotQueryBuilderType() == PostgresConnectorConfig.IncrementalSnapshotChunkQueryBuilderType.UNION_BASED) {
+            return buildUnionBasedChunkQuery(table, limit, additionalCondition);
+        }
+
+        return super.buildChunkQuery(table, limit, additionalCondition);
+    }
+
+    @Override
+    protected PreparedStatement readTableChunkStatement(String sql) throws SQLException {
+        final PreparedStatement statement = jdbcConnection.readTablePreparedStatement(connectorConfig, sql,
+                OptionalLong.empty());
+        if (context.isNonInitialChunk()) {
+            final Object[] maximumKey = context.maximumKey().get();
+            final Object[] chunkEndPosition = context.chunkEndPosititon();
+            // Fill boundaries placeholders
+            int pos = 0;
+            final List<Column> queryColumns = getQueryColumns(getCurrentTable());
+
+            if (postgresConnectorConfig.getIncrementalSnapshotQueryBuilderType().equals(PostgresConnectorConfig.IncrementalSnapshotChunkQueryBuilderType.SIMPLE)) {
```

Comment:
Do we need to copy paste this code from parent class? Can we just call `super.readTableChunkStatement` in case of `SIMPLE` builder?